### PR TITLE
Update port and queue stat counters by Flex counter

### PR DESCRIPTION
**Pull request by andrii-savka**
Updated the implementation to use Flex counter for review. Under manual testing.

### CODE REVIEW

**Comment by andrii-savka** [comment]
Please, review and merge below first:
https://github.com/Azure/sonic-sairedis/pull/230

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -58,7 +59,10 @@ PortsOrch::PortsOrch(DBConnector *db, vector<string> tableNames) :
     /* Initialize queue tables */
     m_queueTable = unique_ptr<Table>(new Table(counter_db, COUNTERS_QUEUE_NAME_MAP));
     m_queuePortTable = unique_ptr<Table>(new Table(counter_db, COUNTERS_QUEUE_PORT_MAP));
-    m_queueIndexTable = unique_ptr<Table>(new Table(counter_db, COUNTERS_QUEUE_INDEX_MAP));
+    m_queueIndexTable = unique_ptr<Table>(new Table(counter_db, COUNTERS_QUEUE_INDEX_MAP));    
+
+    DBConnector *flex_db(new DBConnector(PFC_WD_DB, DBConnector::DEFAULT_UNIXSOCKET, 0));
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >flex_db [](start = 17, length = 7)

memory leak

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -616,9 +620,29 @@ bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
...
-                vector.push_back(tuple);
-                m_counterTable->set("", vector);
+                vector<FieldValueTuple> fields;
+                fields.push_back(tuple);
+                m_counterTable->set("", fields);
+
+                /* Add port to flex_counter for updating stat counters  */
+                string key = sai_serialize_object_id(p.m_port_id) + ":" + FLEX_STAT_COUNTER_POLL_MSECS;
+
+                string counters;
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >string [](start = 16, length = 6)

use ostringsteam?

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -616,9 +620,29 @@ bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
...
+                for (int cntr = SAI_PORT_STAT_IF_IN_OCTETS; cntr <= SAI_PORT_STAT_PFC_7_ON2OFF_RX_PKTS; ++cntr)
+                {
+                    counters += sai_serialize_port_stat(static_cast<sai_port_stat_t>(cntr)) + ",";
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >"," [](start = 94, length = 3)

Could you add separator conservatively? ie. no more after the last one.

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -1307,6 +1332,25 @@ void PortsOrch::initializeQueues(Port &port)
                 sai_serialize_object_id(port.m_queue_ids[queueIndex]),
                 to_string(queueIndex));
         queueIndexVector.push_back(queueIndexTuple);
+
+        string key = sai_serialize_object_id(port.m_queue_ids[queueIndex]) + ":" + FLEX_STAT_COUNTER_POLL_MSECS;
+
+        string counters;
+        for (int cntr = SAI_QUEUE_STAT_PACKETS; cntr <= SAI_QUEUE_STAT_SHARED_WATERMARK_BYTES ; ++cntr)
+        {
+            counters += sai_serialize_queue_stat(static_cast<sai_queue_stat_t>(cntr)) + ",";
```
Comment (qiluo-msft) [inline orchagent/portsorch.cpp]: >"," [](start = 88, length = 3)

The same

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (andrii-savka) [reply orchagent/portsorch.cpp]: Fixed for both m_flexCounterTable and m_counterTable

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (andrii-savka) [reply orchagent/portsorch.cpp]: ok.

File: orchagent/portsorch.cpp (hunk 3 above)
Comment (andrii-savka) [reply orchagent/portsorch.cpp]: refactored.

File: orchagent/portsorch.cpp (hunk 4 above)
Comment (andrii-savka) [reply orchagent/portsorch.cpp]: refactored

**Pull request merged**